### ISSUE TITLE
Unexpected Crash 1.20.1

### ISSUE THREAD

**Issue by gabrielchoong** [juliand665/Dynamic-FPS#218]
Hi, I'm not sure if this is relevant, but I just faced a crash when running the latest supported Dynamic FPS mod for Minecraft 1.20.1.

The crash report:

```txt
---- Minecraft Crash Report ----
// Hi. I'm Minecraft, and I'm a crashaholic.

Time: 2024-09-05 17:42:39
Description: Unexpected error

java.lang.NoSuchMethodError: 'net.minecraft.class_340 net.minecraft.class_310.method_53526()'
	at dynamic_fps.impl.util.HudInfoRenderer.drawBatteryOverlay(HudInfoRenderer.java:42)
	at dynamic_fps.impl.util.HudInfoRenderer.renderInfo(HudInfoRenderer.java:25)
	at net.minecraft.class_329.handler$zjj000$dynamic_fps$renderSavingIndicator(class_329.java:2877)
	at net.minecraft.class_329.method_1753(class_329.java:368)
	at net.minecraft.class_757.method_3192(class_757.java:926)
	at net.minecraft.class_310.method_1523(class_310.java:1219)
	at net.minecraft.class_310.method_1514(class_310.java:802)
	at net.minecraft.client.main.Main.main(Main.java:250)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:480)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Render thread
Stacktrace:
	at dynamic_fps.impl.util.HudInfoRenderer.drawBatteryOverlay(HudInfoRenderer.java:42)
	at dynamic_fps.impl.util.HudInfoRenderer.renderInfo(HudInfoRenderer.java:25)
	at net.minecraft.class_329.handler$zjj000$dynamic_fps$renderSavingIndicator(class_329.java:2877)
	at net.minecraft.class_329.method_1753(class_329.java:368)

-- Affected level --
Details:
	All players: 1 total; [class_746['Wycliffed'/227, l='ClientLevel', x=1.50, y=84.49, z=48.50]]
	Chunk stats: 4096, 619
	Level dimension: minecraft:overworld
	Level spawn location: World: (0,63,48), Section: (at 0,15,0 in 0,3,3; chunk contains blocks 0,-64,48 to 15,319,63), Region: (0,0; contains chunks 0,0 to 31,31, blocks 0,-64,0 to 511,319,511)
	Level time: 8504 game time, 8504 day time
	Server brand: fabric
	Server type: Integrated singleplayer server
Stacktrace:
	at net.minecraft.class_638.method_8538(class_638.java:458)
	at net.minecraft.class_310.method_1587(class_310.java:2406)
	at net.minecraft.class_310.method_1514(class_310.java:826)
	at net.minecraft.client.main.Main.main(Main.java:250)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:480)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)

-- Last reload --
Details:
	Reload number: 1
	Reload reason: initial
	Finished: Yes
	Packs: vanilla, fabric, file/Simple Wool, file/sus_sand_1

-- System Details --
Details:
	Minecraft Version: 1.20.1
	Minecraft Version ID: 1.20.1
	Operating System: Mac OS X (aarch64) version 14.6.1
	Java Version: 17.0.8, Microsoft
	Java VM Version: OpenJDK 64-Bit Server VM (mixed mode), Microsoft
	Memory: 489652136 bytes (466 MiB) / 1342177280 bytes (1280 MiB) up to 2147483648 bytes (2048 MiB)
	CPUs: 8
	Processor Vendor: Apple Inc.
	Processor Name: Apple M1
	Identifier: Apple Inc. Family 0x1b588bb3 Model 0 Stepping 0
	Microarchitecture: ARM64 SoC: Firestorm + Icestorm
	Frequency (GHz): 3.20
	Number of physical packages: 1
	Number of physical CPUs: 8
	Number of logical CPUs: 8
	Graphics card #0 name: Apple M1
	Graphics card #0 vendor: Apple (0x106b)
	Graphics card #0 VRAM (MB): 0.00
	Graphics card #0 deviceId: unknown
	Graphics card #0 versionInfo: unknown
	Memory slot #0 capacity (MB): 0.00
	Memory slot #0 clockSpeed (GHz): 0.00
	Memory slot #0 type: unknown
	Virtual memory max (MB): 10240.00
	Virtual memory used (MB): 6774.25
	Swap memory total (MB): 2048.00
	Swap memory used (MB): 1006.69
	JVM Flags: 7 total; -Xmx2G -XX:+UnlockExperimentalVMOptions -XX:+UseG1GC -XX:G1NewSizePercent=20 -XX:G1ReservePercent=20 -XX:MaxGCPauseMillis=50 -XX:G1HeapRegionSize=32M
	Fabric Mods: 
		badoptimizations: BadOptimizations 2.1.4
		create: Create 0.5.1-f-build.1417+mc1.20.1
			com_electronwill_night-config_core: core 3.6.6
			com_electronwill_night-config_toml: toml 3.6.6
			com_google_code_findbugs_jsr305: jsr305 3.0.2
			flywheel: Flywheel 0.6.10-2
			forgeconfigapiport: Forge Config API Port 8.0.0
			milk: Milk Lib 1.2.60
				dripstone_fluid_lib: Dripstone Fluid Lib 3.0.2
			porting_lib_accessors: Porting Lib Accessors 2.3.2+1.20.1
			porting_lib_base: Porting Lib Base 2.3.2+1.20.1
				porting_lib_config: Porting Lib Config 2.3.2+1.20.1
				porting_lib_lazy_registration: Porting Lib Lazy Register 2.3.2+1.20.1
				porting_lib_mixin_extensions: Porting Lib Mixin Extensions 2.3.2+1.20.1
				porting_lib_utility: Porting Lib Utility 2.3.2+1.20.1
			porting_lib_brewing: Porting Lib Brewing 2.3.2+1.20.1
			porting_lib_client_events: Porting Lib Client Events 2.3.2+1.20.1
			porting_lib_entity: Porting Lib Entity 2.3.2+1.20.1
			porting_lib_extensions: Porting Lib Extensions 2.3.2+1.20.1
				porting_lib_attributes: Porting Lib Attributes 2.3.2+1.20.1
				porting_lib_common: Porting Lib Common 2.3.2+1.20.1
			porting_lib_models: Porting Lib Models 2.3.2+1.20.1
			porting_lib_networking: Porting Lib Networking 2.3.2+1.20.1
			porting_lib_obj_loader: Porting Lib Obj Loader 2.3.2+1.20.1
				porting_lib_model_loader: Porting Lib Model Loader 2.3.2+1.20.1
			porting_lib_tags: Porting Lib Tags 3.0
			porting_lib_tool_actions: Porting Lib Tool Actions 2.3.2+1.20.1
			porting_lib_transfer: Porting Lib Transfer 2.3.2+1.20.1
				porting_lib_core: Porting Lib Core 2.3.2+1.20.1
				porting_lib_fluids: Porting Lib Fluids 2.3.2+1.20.1
			reach-entity-attributes: Reach Entity Attributes 2.4.0
			registrate-fabric: Registrate for Fabric 1.3.62-MC1.20.1
				porting_lib_data: Porting Lib Data 2.1.1090+1.20
				porting_lib_model_generators: Porting Lib Model Generators 2.1.1090+1.20
					porting_lib_gametest: Porting Lib GameTest 2.1.1090+1.20
					porting_lib_model_materials: Porting Lib Model Materials 2.1.1090+1.20
		dynamic_fps: Dynamic FPS 3.6.3
			net_lostluma_battery: battery 1.1.0
		dynamiclights: Dynamic Lights 1.8.3+mod
		fabric-api: Fabric API 0.92.2+1.20.1
			fabric-api-lookup-api-v1: Fabric API Lookup API (v1) 1.6.36+1802ada577
			fabric-biome-api-v1: Fabric Biome API (v1) 13.0.13+1802ada577
			fabric-block-api-v1: Fabric Block API (v1) 1.0.11+1802ada577
			fabric-blockrenderlayer-v1: Fabric BlockRenderLayer Registration (v1) 1.1.41+1802ada577
			fabric-client-tags-api-v1: Fabric Client Tags 1.1.2+1802ada577
			fabric-command-api-v1: Fabric Command API (v1) 1.2.34+f71b366f77
			fabric-command-api-v2: Fabric Command API (v2) 2.2.13+1802ada577
			fabric-commands-v0: Fabric Commands (v0) 0.2.51+df3654b377
			fabric-containers-v0: Fabric Containers (v0) 0.1.64+df3654b377
			fabric-content-registries-v0: Fabric Content Registries (v0) 4.0.11+1802ada577
			fabric-convention-tags-v1: Fabric Convention Tags 1.5.5+1802ada577
			fabric-crash-report-info-v1: Fabric Crash Report Info (v1) 0.2.19+1802ada577
			fabric-data-attachment-api-v1: Fabric Data Attachment API (v1) 1.0.0+de0fd6d177
			fabric-data-generation-api-v1: Fabric Data Generation API (v1) 12.3.4+1802ada577
			fabric-dimensions-v1: Fabric Dimensions API (v1) 2.1.54+1802ada577
			fabric-entity-events-v1: Fabric Entity Events (v1) 1.6.0+1c78457f77
			fabric-events-interaction-v0: Fabric Events Interaction (v0) 0.6.2+1802ada577
			fabric-events-lifecycle-v0: Fabric Events Lifecycle (v0) 0.2.63+df3654b377
			fabric-game-rule-api-v1: Fabric Game Rule API (v1) 1.0.40+1802ada577
			fabric-item-api-v1: Fabric Item API (v1) 2.1.28+1802ada577
			fabric-item-group-api-v1: Fabric Item Group API (v1) 4.0.12+1802ada577
			fabric-key-binding-api-v1: Fabric Key Binding API (v1) 1.0.37+1802ada577
			fabric-keybindings-v0: Fabric Key Bindings (v0) 0.2.35+df3654b377
			fabric-lifecycle-events-v1: Fabric Lifecycle Events (v1) 2.2.22+1802ada577
			fabric-loot-api-v2: Fabric Loot API (v2) 1.2.1+1802ada577
			fabric-loot-tables-v1: Fabric Loot Tables (v1) 1.1.45+9e7660c677
			fabric-message-api-v1: Fabric Message API (v1) 5.1.9+1802ada577
			fabric-mining-level-api-v1: Fabric Mining Level API (v1) 2.1.50+1802ada577
			fabric-model-loading-api-v1: Fabric Model Loading API (v1) 1.0.3+1802ada577
			fabric-models-v0: Fabric Models (v0) 0.4.2+9386d8a777
			fabric-networking-api-v1: Fabric Networking API (v1) 1.3.11+1802ada577
			fabric-networking-v0: Fabric Networking (v0) 0.3.51+df3654b377
			fabric-object-builder-api-v1: Fabric Object Builder API (v1) 11.1.3+1802ada577
			fabric-particles-v1: Fabric Particles (v1) 1.1.2+1802ada577
			fabric-recipe-api-v1: Fabric Recipe API (v1) 1.0.21+1802ada577
			fabric-registry-sync-v0: Fabric Registry Sync (v0) 2.3.3+1802ada577
			fabric-renderer-api-v1: Fabric Renderer API (v1) 3.2.1+1802ada577
			fabric-renderer-indigo: Fabric Renderer - Indigo 1.5.2+85287f9f77
			fabric-renderer-registries-v1: Fabric Renderer Registries (v1) 3.2.46+df3654b377
			fabric-rendering-v0: Fabric Rendering (v0) 1.1.49+df3654b377
			fabric-rendering-v1: Fabric Rendering (v1) 3.0.8+1802ada577
			fabric-resource-conditions-api-v1: Fabric Resource Conditions API (v1) 2.3.8+1802ada577
			fabric-screen-api-v1: Fabric Screen API (v1) 2.0.8+1802ada577
			fabric-screen-handler-api-v1: Fabric Screen Handler API (v1) 1.3.30+1802ada577
			fabric-sound-api-v1: Fabric Sound API (v1) 1.0.13+1802ada577
			fabric-transfer-api-v1: Fabric Transfer API (v1) 3.3.5+8dd72ea377
			fabric-transitive-access-wideners-v1: Fabric Transitive Access Wideners (v1) 4.3.1+1802ada577
		fabricloader: Fabric Loader 0.16.4
			mixinextras: MixinExtras 0.4.1
		ferritecore: FerriteCore 6.0.1
		immediatelyfast: ImmediatelyFast 1.2.21+1.20.4
			net_lenni0451_reflect: Reflect 1.3.4
		indium: Indium 1.0.34+mc1.20.1
		java: OpenJDK 64-Bit Server VM 17
		lithium: Lithium 0.11.2
		macos_input_fixes: MacOS Input Fixes 1.7
		memoryleakfix: Memory Leak Fix 1.1.5
		minecraft: Minecraft 1.20.1
		modernfix: ModernFix 5.19.4+mc1.20.1
		modmenu: Mod Menu 7.2.2
		notenoughcrashes: Not Enough Crashes 4.4.7+1.20.1
		railways: Create: Steam 'n' Rails 1.6.4+fabric-mc1.20.1
			mm: Manningham Mills 2.3
		sodium: Sodium 0.5.11+mc1.20.1
			fabric-api-base: Fabric API Base 0.4.31+1802ada577
			fabric-block-view-api-v2: Fabric BlockView API (v2) 1.0.1+1802ada577
			fabric-rendering-data-attachment-v1: Fabric Rendering Data Attachment (v1) 0.3.37+92a0d36777
			fabric-rendering-fluids-v1: Fabric Rendering Fluids (v1) 3.0.28+1802ada577
			fabric-resource-loader-v0: Fabric Resource Loader (v0) 0.11.10+1802ada577
		sodium-extra: Sodium Extra 0.5.4+mc1.20.1-build.115
			caffeineconfig: CaffeineConfig 1.3.0+1.17
			crowdin-translate: CrowdinTranslate 1.4+1.19.3
	Flywheel Backend: GL33 Instanced Arrays
	Launched Version: fabric-loader-0.16.4-1.20.1
	Backend library: LWJGL version 3.3.1 SNAPSHOT
	Backend API: Apple M1 GL version 4.1 Metal - 88.1, Apple
	Window size: 1440x900
	GL Caps: Using framebuffer using OpenGL 3.2
	GL debug messages: <disabled>
	Using VBOs: Yes
	Is Modded: Definitely; Client brand changed to 'fabric'; Server brand changed to 'fabric'
	Type: Integrated Server (map_client.txt)
	GPU Warnings: version: Metal
	Graphics mode: fast
	Resource Packs: vanilla, fabric, file/Simple Wool, file/sus_sand_1
	Current Language: en_us
	CPU: 8x Apple M1
	Server Running: true
	Player Count: 1 / 8; [class_3222['Wycliffed'/227, l='ServerLevel[trains]', x=1.50, y=84.49, z=48.50]]
	Data Packs: vanilla, fabric
	Enabled Feature Flags: minecraft:vanilla
	World Generation: Stable
	Client Crashes Since Restart: 1
	Integrated Server Crashes Since Restart: 0
	Suspected Mods: Minecraft (minecraft), Fabric Loader (fabricloader), Dynamic FPS (dynamic_fps)
```

It encouraged me to submit a report and here it is!

**Comment by gabrielchoong** [comment]
This crash happened when I tried to load into a single player world, not when launching a minecraft instance. I will try different versions of the mod and see if there are any differences.

**Comment by LostLuma** [comment]
This should happen with any version of the mod that has the battery-related features on 1.20.1.
What I'd recommend is that you disable the entire Battery Tracking feature in the config before playing while this is broken.

I can see if I have time to work on this today or tomorrow, but the fix should be super straightforward so it shouldn't take too long.

**Comment by LostLuma** [comment]
Hello, I finally got around to fixing this.
[dynamic-fps-3.7.1+minecraft-1.20.0-fabric.jar](https://files.lostluma.net/dynamic-fps-3.7.1+minecraft-1.20.0-fabric.jar)

I'm currently sick and get motion sick super fast so I did not do any testing.
Please let me know if the jar above works as expected, then I will also release it on Modrinth etc.

**Comment by Pixaurora** [comment]
It works on my end!

**Comment by LostLuma** [comment]
Thanks :) This is now fixed in [`v3.7.1`](https://github.com/juliand665/Dynamic-FPS/releases/tag/3.7.1). CurseForge release will be later as it's down at this time.

**Comment by gabrielchoong** [comment]
Works for me too. Thanks for you work!
